### PR TITLE
Update `lightning.gpu` to build and run with the recent changes in `lightning.qubit` 

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
This PR, 
- Updates GCC version and moves to C++20
- Updates (Test_)StateVectorCuda* w.r.t new kernel dispatch mechanism in Lightning
- Updates handling statevector data types in Python

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,7 +98,7 @@ set(ENABLE_OPENMP ON)
 FetchContent_Declare(
     pennylane_lightning
     GIT_REPOSITORY https://github.com/PennyLaneAI/pennylane-lightning.git
-    GIT_TAG        latest
+    GIT_TAG        e4bca20acc8c3f2c59f1cd075abd57f7c78de23f # latest
```
Comment (maliasadi) [inline CMakeLists.txt:101]: It will be updated.

File: tests/test_sample.py [hunk 2]
```diff
@@ -92,11 +92,3 @@ def test_sample_values(self, dev, C, tol):
         # s1 should only contain 1 and -1, which is guaranteed if
         # they square to 1
         assert np.allclose(s1**2, 1, atol=tol, rtol=0)
-
-    def test_sample_unsupported_type(self, dev):
```
Comment (maliasadi) [inline tests/test_sample.py:96]: No need for this one anymore.

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 3]
```diff
@@ -389,9 +381,6 @@ def var(self, observable, shot_range=None, bin_size=None):
 
         return squared_mean - (mean**2)
 
-    def sample(self, **kwargs):
```
Comment (maliasadi) [inline pennylane_lightning_gpu/lightning_gpu.py:392]: Hi @mlxd. I suppose this no longer is required. right?

Comment (mlxd) [reply pennylane_lightning_gpu/lightning_gpu.py:392]: @trevor-vincent  I'll defer to you?

File: tests/test_sample.py (hunk 2 above)
Comment (mlxd) [reply tests/test_sample.py:96]: @trevor-vincent And here

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 3 above)
Comment (trevor-vincent) [reply pennylane_lightning_gpu/lightning_gpu.py:376]: Yep not required!

**Review by trevor-vincent — APPROVED** [review]
Nice work Ali!

**Review by mlxd — APPROVED** [review]
Nice work @maliasadi

**Comment by mlxd** [comment]
Also, should we be merging this onto the RC branch, rather than main?

**Review by mlxd — APPROVED** [review]

**Pull request merged**